src/call/github-events-to-s3.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+// Name         : githubEventsToS3
+// Description  : Stores GitHub Events in an S3 Bucket
+
+import { Probot } from 'probot';
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+import { Resource } from '../service/resource/resource';
+import { validateResourceConfig } from '../utility/verification/verify-resource';
+
+export default async function githubEventsToS3(app: Probot, context: any, resource: Resource): Promise<void> {
+  if (!(await validateResourceConfig(app, context, resource))) return;
+
+  const now = new Date();
+  const [day, month, year] = [now.getDate(), now.getMonth() + 1, now.getFullYear()].map((num) => String(num).padStart(2, '0'));
+
+  try {
+    const s3Client = new S3Client({ region: String(process.env.REGION) });
+    const putObjectCommand = new PutObjectCommand({
+      Bucket: 'github-events',
+      Body: JSON.stringify(context),
+      Key: `${year}-${month}-${day}/${context.name}/${context.id}`,
```

Comment:
Can you please share a sample events and its s3 prefix ?

Reply:
A sample event would look like:
{
  name: 'issues',
  id: '54518230-811c-11ef-8496-ec9c5edfd972',
  payload: {
     ...
  },
}

And the s3 prefix would be: s3://github-events/2024-10-08/issues/54518230-811c-11ef-8496-ec9c5edfd972

Reply:
Option 1: With `s3://github-events/2024-10-08/issues/54518230-811c-11ef-8496-ec9c5edfd972`
### Advantages:
- Time-based access: Storing events by date first makes it easier to query or list events for a specific day. If your use case is more focused on querying by date or creating reports based on time, this structure is beneficial.
- Scalability: Grouping by date reduces the number of items in a single folder, which can improve performance with large datasets.
### Disadvantages:
If you need to query all events of a particular type (e.g., issues), this format makes it a little more difficult because the event type (issues) comes after the date.

Option 2: with `s3://github-events/issues/2024-10-08/54518230-811c-11ef-8496-ec9c5edfd972`
### Advantages:
- Event type-based access: Storing by event type first makes it easier to query all events of a certain type (e.g., all issues), which is helpful if you're looking to group events based on their type.
- Organized by type: If you handle multiple event types like issues, pull_requests, etc., this format helps you keep them better organized.
### Disadvantages:
Querying or grouping based on date would be slightly less efficient since it's nested under the event type.

Reply:
Changed the PR to match option 2